hack/cluster-deploy.sh

```diff
@@ -31,10 +31,29 @@ function main() {
 
     # Ignore errors because some clusters might not have prometheus operator
     kubectl apply -f ${MANIFESTS_OUT_DIR} || true
-    kubectl rollout status daemonset kepler-exporter -n kepler --timeout 60s
-
-    echo "Check the logs of the kepler-exporter with:"
-    echo "kubectl -n kepler logs daemonset.apps/kepler-exporter"
+    
+    # round for 3 times and each for 60s
+    # check if the rollout status is running
+    deploy_status=1
+    for i in 1 2 3
+    do
+        echo "check deployment status for round $i"
+        kubectl rollout status daemonset kepler-exporter -n kepler --timeout 60s
+        #check rollout status
+        if [ $? -eq 0 ]
+        then
+            deploy_status=0
+            break
+        fi
+    done 
+    # if deployment in error
+    if test $[deploy_status] -eq 1
```

Comment:
Why don't use the TRAP command?
So, if the error happens in any other place in the script, we will dump the logs

Reply:
may I have a link for TRAP command for details?

Reply:
--> [https://medium.com/@dirk.avery/the-bash-trap-trap-ce6083f36700](https://medium.com/@dirk.avery/the-bash-trap-trap-ce6083f36700)